src/rational/mat_q/properties.rs

```diff
@@ -0,0 +1,134 @@
+// Copyright © 2023 Phil Milewski
+//
+// This file is part of qFALL-math.
+//
+// qFALL-math is free software: you can redistribute it and/or modify it under
+// the terms of the Mozilla Public License Version 2.0 as published by the
+// Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
+
+//! This module includes functionality about properties of [`MatQ`] instances.
+
+use super::MatQ;
+use flint_sys::fmpq_mat::{fmpq_mat_is_one, fmpq_mat_is_square, fmpq_mat_is_zero};
+
+impl MatQ {
+    /// Checks if a [`MatQ`] is the identity matrix.
+    ///
+    /// Returns true if every diagonal entry is `1` and all other entries are
+    /// `0`.
+    ///
+    /// ```
+    /// use qfall_math::rational::MatQ;
+    /// use std::str::FromStr;
+    ///
+    /// let value = MatQ::from_str("[[1, 0],[0, 1]]").unwrap();
+    /// assert!(value.is_identity())
+    /// ```
+    pub fn is_identity(&self) -> bool {
+        1 == unsafe { fmpq_mat_is_one(&self.matrix) }
+    }
+
+    /// Checks if a [`MatQ`] is a square matrix.
+    ///
+    /// Returns true if the number of rows and columns is identical.
+    ///
+    /// ```
+    /// use qfall_math::rational::MatQ;
+    /// use std::str::FromStr;
+    ///
+    /// let value = MatQ::from_str("[[4/7, 0],[5/8, 1/9]]").unwrap();
+    /// assert!(value.is_square())
+    /// ```
+    pub fn is_square(&self) -> bool {
+        1 == unsafe { fmpq_mat_is_square(&self.matrix) }
+    }
+
+    /// Checks if every entry of a [`MatQ`] is `0`.
+    ///
+    /// Returns true if every entry is `0`.
+    ///
+    /// ```
+    /// use qfall_math::rational::MatQ;
+    /// use std::str::FromStr;
+    ///
+    /// let value = MatQ::from_str("[[0, 0],[0, 0]]").unwrap();
+    /// assert!(value.is_zero())
+    /// ```
+    pub fn is_zero(&self) -> bool {
+        1 == unsafe { fmpq_mat_is_zero(&self.matrix) }
+    }
+}
+
+#[cfg(test)]
+mod test_is_identity {
+    use super::MatQ;
+    use std::str::FromStr;
+
+    /// ensure that is_identity returns `true` for identity matrices
```

Comment:
```suggestion
    /// Ensure that is_identity returns `true` for identity matrices.
```
Start with upper case and end with a dot as discussed today. Also applies to the next comments.

Reply:
Optionally also backticks around the function name.
```suggestion
    /// Ensure that `is_identity` returns `true` for identity matrices.
```